main.go

```diff
@@ -45,580 +35,83 @@ var (
 	maxOpenConns       = kingpin.Flag("database.maxOpenConns", "Number of maximum open connections in the connection pool. (env: DATABASE_MAXOPENCONNS)").Default(getEnv("DATABASE_MAXOPENCONNS", "10")).Int()
 	tlsconfigFile      = kingpin.Flag("web.config", "Path to config yaml file that can enable TLS or authentication.").Default("").String()
 	scrapeInterval     = kingpin.Flag("scrape.interval", "Interval between each scrape. Default is to scrape on collect requests").Default("0s").Duration()
+	gracefulStop       = make(chan os.Signal)
 )
 
-// Metric name parts.
-const (
-	namespace = "oracledb"
-	exporter  = "exporter"
-)
-
-// Metrics object description
-type Metric struct {
-	Context          string
-	Labels           []string
-	MetricsDesc      map[string]string
-	MetricsType      map[string]string
-	MetricsBuckets   map[string]map[string]string
-	FieldToAppend    string
-	Request          string
-	IgnoreZeroResult bool
-}
-
-// Used to load multiple metrics from file
-type Metrics struct {
-	Metric []Metric
-}
-
-// Metrics to scrap. Use external file (default-metrics.toml and custom if provided)
-var (
-	metricsToScrap    Metrics
-	additionalMetrics Metrics
-	hashMap           map[int][]byte
-)
+func main() {
+	promLogConfig := &promlog.Config{}
 
-// Exporter collects Oracle DB metrics. It implements prometheus.Collector.
-type Exporter struct {
-	dsn             string
-	duration, error prometheus.Gauge
-	totalScrapes    prometheus.Counter
-	scrapeErrors    *prometheus.CounterVec
-	scrapeResults   []prometheus.Metric
-	up              prometheus.Gauge
-	db              *sql.DB
-	logger          log.Logger
-}
+	flag.AddFlags(kingpin.CommandLine, promLogConfig)
+	kingpin.HelpFlag.Short('\n')
+	kingpin.Version(version.Print("oracledb_exporter"))
+	kingpin.Parse()
+	logger := promlog.New(promLogConfig)
 
-// getEnv returns the value of an environment variable, or returns the provided fallback value
-func getEnv(key, fallback string) string {
-	if value, ok := os.LookupEnv(key); ok {
-		return value
-	}
-	return fallback
-}
+	signal.Notify(gracefulStop, syscall.SIGTERM)
+	signal.Notify(gracefulStop, syscall.SIGINT)
+	signal.Notify(gracefulStop, syscall.SIGHUP)
+	signal.Notify(gracefulStop, syscall.SIGQUIT)
```

Comment:
This seems like new code beyond the refactor; Is this something we need to add?

Also not sure if the behavior added here is any different in the end from the default handlers here, which I believe just shut down the program.